src/mame/yamaha/ymdx9.cpp

```diff
@@ -0,0 +1,403 @@
+// license:BSD-3-Clause
+// copyright-holders:ajxs
+/*******************************************************************************
+
+    Skeleton driver for the Yamaha DX9 FM synthesizer.
+    There is currently no MAME emulation of the OPS/EGS chips, so emulating
+    the synth's tone generation functionality is not possible.
+    The cassette interface is currently not emulated.
+    While there are rumours that an updated firmware ROM exists, this driver
+    is set up to work with the only one that is widely available.
+
+*******************************************************************************/
+
+#include "emu.h"
+
+#include "bus/midi/midi.h"
+#include "cpu/m6800/m6801.h"
+#include "machine/adc0808.h"
+#include "machine/clock.h"
+#include "machine/nvram.h"
+#include "video/hd44780.h"
+
+#include "emupal.h"
+#include "screen.h"
+
+#include "dx9.lh"
+
+
+namespace {
+
+class yamaha_dx9_state : public driver_device
+{
+public:
+	yamaha_dx9_state(const machine_config &mconfig, device_type type, const char *tag)
+			: driver_device(mconfig, type, tag),
+			m_maincpu(*this, "maincpu"),
+			m_adc(*this, "adc"),
```

Comment:
Also kinda trivial, but you should either go with colon and commas at the end, or at the beginning, not a mixture.

e.g. this:
```
    C(...) :
        m1(...),
        m2(...),
        ...
```

or this:
```
    C(...)
        : m1(...)
        , m2(...)
        ...
```